src/app/app.component.spec.ts

```diff
@@ -3,15 +3,74 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-import { TestBed } from '@angular/core/testing';
-import { AppComponent } from './app.component';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing'
+import { By } from '@angular/platform-browser'
+import { AppComponent } from './app.component'
+import { TestCommon } from './test/testCommon'
 
 describe('AppComponent', () => {
+  // NOTE: If location.href change, can not run test any more,
+  // In other hands, location.href is not configurable so that could not mock the href by spyOnProperty
+  // so here use fakeLocation
+  const fakeLocation = {
+    href: ""
+  }
+
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       declarations: [
         AppComponent
       ],
-    }).compileComponents();
+      providers: [
+        { provide: 'Location', useValue: fakeLocation }
+      ]
+    }).compileComponents()
   });
-});
+
+  it('should display icon and description', fakeAsync(() => {
+    // given
+    const shareUrl = "https://google.com"
+    fakeLocation.href = TestCommon.generateTestShareLink(shareUrl)
+
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+    tick()
+
+    TestCommon.mockUserAgent(TestCommon.USER_AGENT_ANDROID_WEB_VIEW)
+
+    // when
+    app.ngOnInit()
+    fixture.detectChanges()
+
+    // then
+    const icon = fixture.debugElement.query(By.css(".bannerIconLarge"))
+    expect(icon.nativeElement).toBeDefined()
+    const description = fixture.debugElement.query(By.css(".bannerDescription"))
+    expect(description.nativeElement.textContent.trim()).toEqual(app.DEFAULT_BANNER_DESCRIPTION)
+
+    tick(500)
+  }));
+
+  it('should display click banner if sharelink not navigated', fakeAsync(() => {
+    // given
+    const shareUrl = "https://google.com"
+    fakeLocation.href = TestCommon.generateTestShareLink(shareUrl)
+
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+
+    TestCommon.mockUserAgent(TestCommon.USER_AGENT_ANDROID_CHROME_BROWSER)
```

Comment:
Is there any reason why you use TestCommon for the mocking userAgent instead of like the below code?
```
spyOnProperty(navigator, 'userAgent').and.returnValue(TestCommon.USER_AGENT_ANDROID_CHROME_BROWSER)
```

Reply:
It's going to be used many times same code
spyOnProperty(navigator, 'userAgent').and.returnValue(something).
so extracted to utility function